### PR TITLE
nextcloudPackages: init

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Ability to package and install Nextcloud apps like this:
```
  services.nextcloud = {
    enable = true;
    package = pkgs.nextcloud25;
    hostName = "localhost";
    config.adminpassFile = "${pkgs.writeText "adminpass" "hunter2"}";
    extraApps = with pkgs.nextcloud25Packages.apps; {
      inherit mail news contacts;
    };
    extraAppsEnable = true;
  };
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/nextcloud/packages/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+= Adding plugin, theme or language =
+
+To extend the wordpressPackages set, add a new line to the corresponding json
```
Comment (bjornfor) [inline pkgs/servers/nextcloud/packages/README.md:3]: wordpressPackages -> nextcloudPackages

**Comment by onny** [comment]
Okay it's finally working. The current syntax is a bit different
```
  services.nextcloud = {
    enable = true;
    package = pkgs.nextcloud24;
    hostName = "localhost";
    config.adminpassFile = "${pkgs.writeText "adminpass" "hunter2"}";
    extraApps = with pkgs.nextcloud25Packages.apps; {
      mail = mail;
      calendar = calendar;
      contacts = contacts;
    };
    extraAppsEnable = true;
  };
```
the doubled mention of the app ids if somehow required by the nextcloud module. maybe someone knows a bit cleaner or nicer syntax.

**Comment by bjornfor** [comment]
> the doubled mention of the app ids if somehow required by the nextcloud module. maybe someone knows a bit cleaner or nicer syntax.

What about

```
  services.nextcloud = {
    [...]
    extraApps = with pkgs.nextcloud25Packages.apps; {
      inherit mail calendar contacts;
    };
    extraAppsEnable = true;
  };
```

?

**Comment by onny** [comment]
Oh of course, this works, thank you :)

File: pkgs/servers/nextcloud/packages/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+``` 
+./generate.sh
+``` 
+
+After that you can commit and submit the changes.
+
+= Usage with the Nextcloud module =
+
+The apps will be available in the namespace `nextcloud25Packages.apps`.
```
Comment (SuperSandro2000) [inline pkgs/servers/nextcloud/packages/README.md:24]: Why though? Whats under nextcloud25Packages?

File: pkgs/servers/nextcloud/packages/generate.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -I nixpkgs=../../../.. -i bash -p nc4nix
+
+set -e
+set -u
+set -o pipefail
+set -x
```
Comment (SuperSandro2000) [inline pkgs/servers/nextcloud/packages/generate.sh:7]:
```suggestion
set -euo pipefail
```

turning on -x by default can quite spam the terminal log.

File: pkgs/servers/nextcloud/packages/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+- `nextcloud-apps.json` for apps
+
+The app must be available in the official
+[Nextcloud app store](https://apps.nextcloud.com).
+https://apps.nextcloud.com. The id corresponds to the last part in the app url,
+for example `breezedark` for the app with the url
+`https://apps.nextcloud.com/apps/breezedark`.
+
+To regenerate the nixpkgs nextcloudPackages set, run:
```
Comment (SuperSandro2000) [inline pkgs/servers/nextcloud/packages/README.md:14]: Why do we not generate this from the webstore completely automated?

**Pull request merged**